### DISCUSSION TITLE
I create a chaos that runs continuously, then suspend it and delete the pod, deployment will regenerate a pod, and then start the chaos again, there will be a problem that the previous pod cannot be found, and the Chaos fault cannot be injected, how can the controller automatically discover the new pod ?

### DISCUSSION THREAD

**Discussion by AllenZMC** [chaos-mesh/chaos-mesh#3690] [Q&A] [answered]
I create a chaos that runs continuously, then suspend it and delete the pod, deployment will regenerate a pod, and then start the chaos again, there will be a problem that the previous pod cannot be found, and the Chaos fault cannot be injected, how can the controller automatically discover the new pod ?

**Comment by STRRL** [comment, marked as answer]
Chaos Mesh could ONLY inject chaos into existing pods, and it could NOT inject chaos "dynamically" into the new pods now. :cry:

And there are some issues/discussions like:

- https://github.com/chaos-mesh/chaos-mesh/issues/1259
- https://github.com/chaos-mesh/rfcs/pull/12

And actually, it's not in active development because there has not  a convincing design for it.

**Comment by jfsmith-at-coveo** [comment]
One workaround I've found is to run a Container Kill PODChaos experiment on the same target at the same time. This reboots the container without destroying the pod.

This hasn't worked with continuous JVM attacks because the injection is removed upon reboot, but I've done it with a continuous DNSChaos attack and it has the intended effect.
